### PR TITLE
Increase transaction size limit for Ledger Nano X

### PR DESCRIPTION
**Pull request by kaifeur**
So as Ledger Nano X has 32K RAM, it allows us to increase the limit of transaction size for this device. 
Reference for this approach was found in [LedgerHQ/app-xrp](https://github.com/LedgerHQ/app-xrp/blob/7562eb56605f954702a9139f235b57ef7f1fb603/src/limitations.h#L35).

### CODE REVIEW

**Comment by frol** [comment]
My only concern is that I am not sure if the apps are compiled differently for Nano S and Nano X, or it is a single binary. If it is a single binary, this compile-time variable is not going to help.

**Comment by frol** [comment]
I have no experience with Ledger devices, so I just raised the concern and indeed, given that those are different SDKs, it seems they get compiled separately. I believe we are blocked on @vgrichina here anyway.

**Comment by kaifeur** [comment]
Some day somebody will review this pull request...

**Comment by volovyks** [comment]
Hey @kaifeur! This app was not under active development, we see your PR and will review it ASAP, but it requires some effort from our side.
The change looks very promising, it should help in a lot of use cases.

**Comment by kaifeur** [comment]
Two weeks seems too long for +13/-0, guys

**Comment by iamsteeve** [comment]
OMG I can't believe this doesn't allow me to do a simple NEAR - REF swap.

**Comment by colinlgray** [comment]
Big thanks to @kaifeur for writing this. I have funds stuck on the Rainbow Bridge and am unable to finalize the transaction because the message is too long. Please @volovyks review this ASAP it is something that I greatly need

**Comment by kaifeur** [comment]
@frol @volovyks @vgrichina It is almost a month and a half for this really tiny PR. Guys please merge it.

**Review by frol — APPROVED** [review]
@janewang The change looks totally fine to me from the code point of view. Yet, I have no idea how to get the app to the Ledger marketplace, so it is still blocked on someone from the tooling team to pick it up

**Review by frol — APPROVED** [review]
@janewang The change looks totally fine to me from the code point of view. Yet, I have no idea how to get the app to the Ledger marketplace, so it is still blocked on someone from the tooling team to pick it up

**Comment by kaifeur** [comment]
@frol it seems like you should remove @volovyks from approvers to merge this pull request.
by the way may be you can contact @volovyks about publishing these changes into the Ledger marketplace? he had merged a pull request into the Ledger repository before.

**Comment by frol** [comment]
I have been in touch with the tooling team inside Pagoda, and @DavidM-D said that he was able to test this patch successfully. He is underwater these days so he will get back as soon as possible and merge this PR & publish the new version to Ledger marketplace.

**Comment by kaifeur** [comment]
@frol @DavidM-D any updates?

**Comment by volovyks** [comment]
@kaifeur I'm on it.

**Review by volovyks — APPROVED** [review]
Tested on Physical Nano S Device. Had some problems testing it on Nano X emulator, probably because our codebase is outdated.
@kaifeur thank you for this contribution and sorry for this huge delay.

**Comment by frol** [comment]
@volovyks Will new users get this updated version on their Ledger devices? Do you need to publish it on Ledger "Store" somehow?

**Comment by volovyks** [comment]
@frol we are not managing the release process. Ledger company is owning the repo and release pipeline.
Here is the PR: https://github.com/LedgerHQ/app-near/pull/18
Now I'm trying to get their attention in Ledger Discord.

**Comment by near-akaia-root** [comment]
[imagen](https://github.com/LedgerHQ/app-near/pull/18#issuecomment-1510911067)

**Pull request merged**
